nest/src/AggregateToken.sol

```diff
@@ -13,7 +14,7 @@ import { IComponentToken } from "./interfaces/IComponentToken.sol";
  * @author Eugene Y. Q. Shen
  * @notice Implementation of the abstract ComponentToken that represents a basket of ComponentTokens
  */
-contract AggregateToken is ComponentToken, IAggregateToken {
+contract AggregateToken is ComponentToken, IAggregateToken, IERC1155Receiver {
```

Comment:
Can you use `ERC1155Holder` instead? https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/utils/ERC1155Holder.sol